src/NuGet.Core/NuGet.LibraryModel/LibraryIdentity.cs

```diff
@@ -40,7 +40,7 @@ public bool Equals(LibraryIdentity other)
             {
                 return true;
             }
-            return string.Equals(Name, other.Name) &&
+            return string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase) &&
                    Equals(Version, other.Version) &&
                    Equals(Type, other.Type);
```

Comment:
Equals calls into the type's overridden `Equals` method. Reference equals is the default implementation but can still be explicitly called with the method `ReferenceEquals`.